File: Introductory Problems/tower of hanoi/tower.js
function towerOfHanoi(n) {
  console.log(2 ** n - 1);
  return function recurse(n, leftStack, rightStack) {
    let midStack = 6 - (leftStack + rightStack);
    if (n == 1) {
      console.log(leftStack, rightStack);
      return;
    }
    midStack = 6 - (leftStack + rightStack);
    recurse(n - 1, leftStack, midStack);
    console.log(leftStack, rightStack);
    recurse(n - 1, midStack, rightStack);
  };
}

towerOfHanoi(2)(2, 1, 3);

File: Introductory Problems/tower of hanoi/tower.py
def tower_of_hanoi(n):
  print(2**n - 1)
  def recurse(n, left_stack, right_stack): 
    mid_stack = 6 - (left_stack + right_stack)
    if n == 1:
        print(left_stack, right_stack)
        return
    mid_stack = 6 - (left_stack + right_stack)
    recurse(n - 1, left_stack, mid_stack)
    print(left_stack, right_stack)
    recurse(n - 1, mid_stack, right_stack)
  return recurse


tower_of_hanoi(2)(2, 1, 3)